### PR TITLE
ipc4: add format conversion support of s32le to s24le

### PR DESCRIPTION
**Pull request by RanderWang**
In ci test a format conversion of S32_LE to
S24_LE is required.

Validated in CI system.

### CODE REVIEW

**Comment by RanderWang** [comment]
@XiaoyunWu6666  please help to validate this PR. Thanks!

File: src/audio/pcm_converter/pcm_converter_generic.c [hunk 1]
```diff
@@ -212,6 +212,35 @@ static int pcm_convert_s32_to_s24(const struct audio_stream *source,
...
+
+	src += ioffset;
+	dst += ooffset;
+	for (processed = 0; processed < samples; processed += n) {
+		src = audio_stream_wrap(source, src);
+		dst = audio_stream_wrap(sink, dst);
+		n = samples - processed;
+		nmax = audio_stream_bytes_without_wrap(source, src) >> BYTES_TO_S32_SAMPLES;
+		n = MIN(n, nmax);
+		nmax = audio_stream_bytes_without_wrap(sink, dst) >> BYTES_TO_S32_SAMPLES;
```
Comment (singalsu) [inline src/audio/pcm_converter/pcm_converter_generic.c:232]: Looks good but you could use here the new function audio_stream_samples_without_wrap_s32() , there's also functions for s24 and s16. It further simplifies the code look.

**Comment by XiaoyunWu6666** [comment]
Thanks for quick response and fix.
Validated that this can fix the error for S24_LE record issues on multiple sof IPC4 platforms

File: src/audio/pcm_converter/pcm_converter_generic.c (hunk 1 above)
Comment (RanderWang) [reply src/audio/pcm_converter/pcm_converter_generic.c]: @singalsu  I updated it , please help to check it . Thanks!

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
one of the DUTs possibly crashed, retry the CI.

**Pull request merged**